fix: replace is-number dependency with a one liner

Abuzeid1 opened the pull request:
This PR replaces is-number package with a one liner with identical code.
This change reduces the unpacked size by a third and making it dependency free.

omrilotan commented on lib/formatter/index.js:
```diff
@@ -1,4 +1,4 @@
-const isNumber = require('is-number');
+const isNumber = (num) => (typeof num === "number" && num - num === 0) || (typeof num === "string" && Number.isFinite(+num) && num.trim() !== "");
```
```suggestion
const isNumber = (num) => typeof num === "number" && num - num === 0;
```

Since we only use "isNumber" for number types — there's no need for the string check. Internally, this can be more efficient
```js
if (typeof value !== 'number' || !isNumber(value)) {
```

omrilotan reviewed (changes requested):
Thank you for your contribution. If you're happy to fulfil the suggestions - I'll gladly merge this fix and release in as a patch.
If you want me to — I can finish this up for you

Abuzeid1 commented:
In the last commit, I removed the string check as  you suggested, then removed the type ```number``` check as it was already checked, what remained was ```num --- num === 0``` which I incorporated into the ```if``` statement.

omrilotan commented:
Thank you. Released with [v1.1.2](https://www.npmjs.com/package/@fiverr/statsd-client/v/1.1.2)
